docs/development/decision-records/2024-08-28-Release-Workflow-Refactor/README.md

```diff
@@ -0,0 +1,83 @@
+
+Github Actions Jobs and Step conditionals are then used to make the execution follow the required steps, as follows.
+
+### Prepare Release workflow logic
+- Triggers manually. Input is the version, choice and base branch
+- if choice is release
+  - branches from main into `release/<input_version>`
+- if choice is bugfix
+  - branches from tag into `bugfix/<input_version>`
+- replaces the project default version `(gradle.properties)` with the workflow version input
+- replaces openapi spec version
+- replaces charts version and README
+- commits and pushes to `release/*` or `bugfix/X.Y.Z` branch
+- if new release
+   - opens PR with new `release/*` branch as head and `main/` as base
+
+Then, a different process takes places if we intend to release a latest version or a bugfix. 
+
+### Case for latest releases
+After the prepare release PR is approved, merging into main will trigger the publish-release workflow where the required publication steps will take place, as shown in [Publish release workflow logic](#publish-release-workflow-logic).
+
+### Case for bugfix Releases
+After the bugfix branch is created, developers commit the fix or set of fixes to it, either via direct push or PR.
+The release of the bugfix version should be triggered manually through the publishing workflow and should execute most of the publication steps as a normal release.
+However, for a bugfix release, the workflow shouldn't:
+- Make any commits to the main branch
+- Bump the project version to the next minor version
+
+### Publish release workflow logic
+- Triggers automatically when prepare release PR is merged into main from `release/*` branch
+- Triggers manually only from branch `bugfix/X.Y.Z`
+
+- Extracts release version from `release/*` or `bugfix/X.Y.Z` branches
+- Triggers release to maven repository by calling `trigger-maven-publish.yaml`
+- Triggers docker image publishing by calling `trigger-maven-publish.yaml`
+- Releases helm-charts
+- Creates Github Tag and Release
+- if PR was merged to main it means a new release then:
+   - Updates project version to next version, commits and pushes to main
+      - if final version, increments to next patch version and adds "-SNAPSHOT"
+      - if not final version (e.g.-rc1), do not increment
+- _(elif PR was merged to hotfix then don't increment)_
+- Publishes to Swaggerhub by calling `publish-swaggerhub.yaml`
+- Publishes OpenAPI UI spec to Github Pages by calling `publish-openapi-ui.yml`
+
+## Other Considerations
+
+- Keeping `release/*` branches is no longer needed since the tag references the commit on main.
+- Release candidates should be handled the same using the same process as a latest release.
+- Release workflows should not run on forks.
+- Should be possible to trigger the publishing of snapshots manually from main or bugfix branches.
+- `draft-new-release` workflow could be named to `manual-release-draft`.
```

Comment:
it could be called `draft-release`, no need to specify `manual` as it's kinda implicit

Reply:
Sugested to name it like that because of what is stated in [this](https://github.com/eclipse-tractusx/tractusx-edc/issues/1445) issue:
_"workflows, that are intended for manual-only tasks should be prefixed manual-, e.g. manual-docker-publish.yaml instead of publish-docker.yaml"_

Reply:
those are workflows, that only do part of a thing, such as publishing docker images. The statement in the issue could have been clearer, sorry.
sometimes manual WFs come in handy, but I agree with @ndr-brt that the actual draft-release workflow does not need it.